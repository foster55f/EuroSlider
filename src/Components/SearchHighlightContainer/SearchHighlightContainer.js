import React from 'react'
import HighlightCard from '../HighlightCard/HighlightCard';
import { connect } from 'react-redux';


export const SearchHighlightContainer = ({ displayGames }) => {
    return (
        <div className= 'game-container'>
            {displayGames.map(game => {
            return (
                <HighlightCard
                key={game.competition.id}
                id={game.competition.id} 
                title={game.title}
                image={game.thumbnail}
                url={game.url}
                embed ={game.videos[0].embed}     
            />
            )
        })}
        </div>
    )
}
export const mapStateToProps = state => ({
    displayGames: state.displayGames
})


export default connect(mapStateToProps)(SearchHighlightContainer)