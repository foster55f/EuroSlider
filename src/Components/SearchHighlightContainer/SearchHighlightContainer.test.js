import React from 'react';
import { shallow } from 'enzyme';
import { SearchHighlightContainer, mapStateToProps } from './SearchHighlightContainer';

describe('SearchHighlightContainer', () => {
  let wrapper;

    beforeEach(() => {
    wrapper = shallow(<SearchHighlightContainer
        displayGames={[{ id: 'barcelona' }]}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

  describe('mapsStateToProps', () => {
    it('should return only game highlights from the store', () => {
        const mockState = {
            displayGames: [{ team: 'barcelona' }],
        };
        const expected = {
            displayGames: [{ team: 'barcelona' }],
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
        });
    });