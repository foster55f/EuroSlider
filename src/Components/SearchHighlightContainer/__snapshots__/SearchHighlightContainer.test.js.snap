// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHighlightContainer should match the snapshot 1`] = `undefined`;
